### PR TITLE
Improve Docker Support

### PR DESCRIPTION
**Pull request by serroba**
# What?

Improve docker support by adding a default working CMD instruction.

By default tries to run the modbus server in a headless mode.

Additionally, adding a bit more details to the documentation

# Why?

This makes the container more useful and friendly. It becomes ready to use with no much more to do.

# Proof it's working

image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # syntax=docker/dockerfile:1
 
-FROM python:3.8-slim-buster
+FROM python:3.9-slim-buster
```
Comment (serroba) [inline Dockerfile:3]: this isn't particularly required, happy to keep in 3.8 if that's a preference

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 
 WORKDIR /pymodbus
 
+EXPOSE 8080
+EXPOSE 5020
+
 COPY . .
 
-RUN pip3 install -r requirements.txt && pip3 install -e .
+RUN pip install -r requirements.txt && pip install -e .
```
[the remark this answers is not in the document]
Comment (janiversen) [reply Dockerfile:12]: I prefer to keep pip3, that way we do not get side effects if python2 is installed.

**Review by janiversen — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (serroba) [reply Dockerfile:12]: Sure I'll update. However, the scenario you are talking about it's not something that would happen accidentally.
Requires modify the dockerfile. However, I don't mind to revert the change

**Review by janiversen — APPROVED** [review]
LGTM.

**Pull request merged**